api/views/assets.py

```diff
@@ -298,7 +300,17 @@ def get_queryset(self):
         return self.queryset.none()
 
     def perform_create(self, serializer):
-        serializer.save(submitted_by=self.request.user)
+        abstract = self.request.FILES.get('police_abstract', None)
+        user = self.request.user
+        if abstract:
+            abstract_name = user_abstract(user, abstract.name)
+            serializer.save(submitted_by=user, police_abstract=abstract_name)
+            fs = FileSystemStorage()
+            filename = fs.save(abstract_name, abstract)
+            uploaded_file_url = fs.url(filename)
+            return Response(data={"success": {"uploaded_file_url": uploaded_file_url}})
```

Comment:
The response seems to not be consistent when police abstract is uploaded and when it's not uploaded. This means when uploaded, the response contains only the uploaded_file_url and when not uploaded, it contains the submitted incident_report fields. I think it would be more appropriate if the uploaded_file_url is the additional field to the response when police abstract is uploaded.